doc/whats_new/upcoming_changes/sklearn.utils/30187.api.rst

```diff
@@ -0,0 +1,4 @@
+- :func:`utils.estimator_checks.check_estimator_sparse_tag` ensures that
+  the estimator tag `input_tags.sparse` is consistent with its `fit`
+  method (acceting sparse input `X` or raising an error).
+  By :user:`Antoine Baker <antoinebaker>`
```

Comment:
Let's include the list of all estimators whose `tags.input_tags.sparse` value was fixed here.

Maybe we could introduce a new subdirectory under `doc/whats_new/upcoming_changes/` dedicated to PRs impacting many modules at once.

WDYT @lesteve @adrinjalali?

Reply:
There is already such a subdirectory named, for instance this particular PR should probably have an entry in `changed-models` to document the fix to all impacted estimators (with the big list) and maybe a second "enhancement" entry under `sklearn.utils` to document the enhancement to the estimator check function.

Reply:
Is that okay that the list will be real big (~100 estimators) ?

Reply:
Indeed if the list that big, it's fine to state that this changes impacts many estimators (even "a majority of estimators" in scikit-learn?)